File: Plugin_en_cours/twitch_hello.py
import socket
import threading
import time
import obspython as obs

# Variables globales pour la configuration
server = "irc.chat.twitch.tv"
port = 6667
nickname = ""
token = ""
channel = ""

irc_socket = None
running = False

def connect_to_twitch():
    global irc_socket, running
    try:
        irc_socket = socket.socket()
        irc_socket.connect((server, port))
        irc_socket.send(f"PASS {token}\r\n".encode('utf-8'))
        irc_socket.send(f"NICK {nickname}\r\n".encode('utf-8'))
        irc_socket.send(f"JOIN #{channel}\r\n".encode('utf-8'))
        obs.script_log(obs.LOG_INFO, "Connecté à Twitch IRC")
    except Exception as e:
        obs.script_log(obs.LOG_ERROR, f"Erreur de connexion : {e}")
        return

    running = True
    while running:
        try:
            response = irc_socket.recv(2048).decode('utf-8')
            if response.startswith("PING"):
                irc_socket.send("PONG :tmi.twitch.tv\r\n".encode('utf-8'))
            else:
                # Vérifie si la commande !hello apparaît dans le message
                if "!hello" in response:
                    obs.script_log(obs.LOG_INFO, "Commande !hello reçue dans le chat!")
        except Exception as e:
            obs.script_log(obs.LOG_ERROR, f"Erreur lors de la réception : {e}")
            break

def start_thread():
    thread = threading.Thread(target=connect_to_twitch)
    thread.daemon = True
    thread.start()

def stop_thread():
    global running
    running = False
    if irc_socket:
        irc_socket.close()

# OBS API : Description du script
def script_description():
    return "Script Python qui se connecte à Twitch et réagit à la commande !hello"

# Mise à jour des paramètres du script
def script_update(settings):
    global nickname, token, channel
    nickname = obs.obs_data_get_string(settings, "nickname")
    token = obs.obs_data_get_string(settings, "token")
    channel = obs.obs_data_get_string(settings, "channel")
    obs.script_log(obs.LOG_INFO, f"Paramètres mis à jour : {nickname}, {channel}")

# Création des propriétés à configurer dans l’interface d’OBS
def script_properties():
    props = obs.obs_properties_create()
    obs.obs_properties_add_text(props, "nickname", "Nickname", obs.OBS_TEXT_DEFAULT)
    obs.obs_properties_add_text(props, "token", "OAuth Token", obs.OBS_TEXT_DEFAULT)
    obs.obs_properties_add_text(props, "channel", "Channel (sans #)", obs.OBS_TEXT_DEFAULT)
    return props

# Chargement et déchargement du script
def script_load(settings):
    start_thread()
    obs.script_log(obs.LOG_INFO, "Script Python chargé et thread démarré")

def script_unload():
    stop_thread()
    obs.script_log(obs.LOG_INFO, "Script Python arrêté")

File: Plugin_en_cours/twitch_hello.lua
local socket = require("socket")

local server = "irc.chat.twitch.tv"
local port = 6667
local nickname = ""
local token = ""
local channel = ""

local irc_socket = nil
local running = false

function connect_to_twitch()
    irc_socket = socket.tcp()
    local success, err = irc_socket:connect(server, port)
    if not success then
        obs.script_log(obs.LOG_ERROR, "Erreur de connexion : " .. tostring(err))
        return
    end
    irc_socket:send("PASS " .. token .. "\r\n")
    irc_socket:send("NICK " .. nickname .. "\r\n")
    irc_socket:send("JOIN #" .. channel .. "\r\n")
    obs.script_log(obs.LOG_INFO, "Connecté à Twitch IRC (Lua)")
    running = true

    while running do
        local response, err = irc_socket:receive("*l")
        if response then
            if string.find(response, "PING") then
                irc_socket:send("PONG :tmi.twitch.tv\r\n")
            elseif string.find(response, "!hello") then
                obs.script_log(obs.LOG_INFO, "Commande !hello reçue dans le chat! (Lua)")
            end
        else
            obs.script_log(obs.LOG_ERROR, "Erreur lors de la réception : " .. tostring(err))
            break
        end
        socket.sleep(0.1)
    end
end

local thread_started = false

-- Pour Lua, nous utilisons un timer OBS afin de lancer la connexion une seule fois.
function start_connection()
    if not thread_started then
        thread_started = true
        -- Lancer la connexion dans une fonction non bloquante via une coroutine
        local co = coroutine.create(connect_to_twitch)
        coroutine.resume(co)
    end
end

function stop_connection()
    running = false
    if irc_socket then
        irc_socket:close()
    end
end

function script_description()
    return "Script Lua qui se connecte à Twitch et réagit à la commande !hello"
end

function script_properties()
    local props = obs.obs_properties_create()
    obs.obs_properties_add_text(props, "nickname", "Nickname", obs.OBS_TEXT_DEFAULT)
    obs.obs_properties_add_text(props, "token", "OAuth Token", obs.OBS_TEXT_DEFAULT)
    obs.obs_properties_add_text(props, "channel", "Channel (sans #)", obs.OBS_TEXT_DEFAULT)
    return props
end

function script_update(settings)
    nickname = obs.obs_data_get_string(settings, "nickname")
    token = obs.obs_data_get_string(settings, "token")
    channel = obs.obs_data_get_string(settings, "channel")
    obs.script_log(obs.LOG_INFO, "Paramètres mis à jour (Lua) : " .. nickname .. ", " .. channel)
end

function script_load(settings)
    -- Lancer la connexion après chargement du script
    start_connection()
    obs.script_log(obs.LOG_INFO, "Script Lua chargé")
end

function script_unload()
    stop_connection()
    obs.script_log(obs.LOG_INFO, "Script Lua arrêté")
end
